assets/js/editor-components/incompatible-payment-gateways-notice/use-incompatible-payment-gateways-notice.ts

```diff
@@ -0,0 +1,95 @@
+/**
+ * External dependencies
+ */
+import { useSelect } from '@wordpress/data';
+import { useEffect, useState } from '@wordpress/element';
+import { useLocalStorageState } from '@woocommerce/base-hooks';
+
+/**
+ * Internal dependencies
+ */
+import { STORE_KEY as PAYMENT_STORE_KEY } from '../../data/payment/constants';
+
+const initialDismissedNotices: React.SetStateAction< object[] > = [];
+
+const areEqual = ( array1: string[], array2: string[] ) => {
```

Comment:
Can we improve this function such that the dismissed notice appears only when a new incompatible gateway is enabled and remains hidden when an incompatible gateway is disabled? Or should we continue as we are (i.e., by enabling or disabling an incompatible gateway the notice will consider this a change of the list of incompatible gateways and will appear again)?